actors/miner/src/lib.rs

```diff
@@ -1950,6 +1958,283 @@ impl Actor {
+        let policy = rt.policy();
+        let miner_id = rt.message().receiver().id().unwrap();
+
+        if !can_prove_commit_ni_seal_proof(rt.policy(), params.seal_proof_type) {
+            return Err(actor_error!(
+                illegal_argument,
+                "unsupported seal proof type {}",
+                i64::from(params.seal_proof_type)
+            ));
+        }
+
+        if params.aggregate_proof_type != RegisteredAggregateProof::SnarkPackV2 {
+            return Err(actor_error!(illegal_argument, "aggregate proof type must be SnarkPackV2"));
+        }
+
+        validate_seal_aggregate_proof(
+            &params.aggregate_proof,
+            params.sectors.len() as u64,
+            policy,
+            true,
+        )?;
+
+        if params.sectors.iter().any(|sector| sector.sealer_id != miner_id) {
+            return Err(actor_error!(
+                illegal_argument,
+                "sealer must be the same as the receiver actor for all sectors"
+            ));
+        }
+
+        if params.sectors.iter().any(|sector| sector.sector_number != sector.sealing_number) {
+            return Err(actor_error!(
+                illegal_argument,
+                "sealing number must be same as sector number for all sectors"
+            ));
+        }
+
+        let curr_epoch = rt.curr_epoch();
+        let challenge_earliest = curr_epoch - rt.policy().max_prove_commit_ni_randomness_lookback;
+        let unsealed_cid = CompactCommD::empty().get_cid(params.seal_proof_type).unwrap();
+        let entropy = serialize(&rt.message().receiver(), "address for get verify info")?;
+
+        for sector in params.sectors.iter() {
+            if !is_sealed_sector(&sector.sealed_cid) {
+                return Err(actor_error!(illegal_argument, "sealed CID had wrong prefix"));
+            }
+
+            if sector.seal_rand_epoch >= curr_epoch {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} must be before now {}",
+                    sector.seal_rand_epoch,
+                    curr_epoch
+                ));
+            }
+
+            if sector.seal_rand_epoch < challenge_earliest {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} too old, must be after {}",
+                    sector.seal_rand_epoch,
+                    challenge_earliest
+                ));
+            }
+        }
+
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let info = get_miner_info(rt.store(), &state)?;
+        let activation_epoch = rt.curr_epoch();
+
+        if params.proving_deadline >= policy.wpost_period_deadlines {
+            return Err(actor_error!(
+                illegal_state,
+                "proving deadline index {} invalid",
+                params.proving_deadline
+            ));
+        }
+
+        if !deadline_is_mutable(
```

Comment:
@anorth @ZenGround0 what will this call do if this is the _first_ time the miner has onboarded sectors so has not enrolled in cron and doesn't have deadlines set up - do we have an initial set of deadlines and will `deadline_is_mutable` do the correct thing?

Reply:
Deadlines and proving period offset are all set up at construction. `state.current_proving_period_start` is explicitly intended to work without an active cron, as a function only of the proving period offset and current epoch.

So this should all work, *but* we are missing an integration test to prove it. What we need is something like batch_onboarding_test, simplified but extended to run one extra proving period and show that a fault is registered if Window PoST is missed. Or, perhaps easier, extend the integration test from this PR to run first a successful Window PoST, then a missed one (see batch_onboarding_test for the flow).

(We are missing this for pcs3 onboarding too. The existing test uses the now-legacy prove_commit_aggregate. But since cron is activated at pre-commit, the interactive prove-commit method used doesn't make a difference here).